crypto/asn1/a_time.c

```diff
@@ -200,16 +393,8 @@ int ASN1_TIME_to_tm(const ASN1_TIME *s, struct tm *tm)
         return 0;
     }
 
-    if (s->type == V_ASN1_UTCTIME) {
-        memset(tm, 0, sizeof(*tm));
-        return asn1_utctime_to_tm(tm, s);
-    }
-    if (s->type == V_ASN1_GENERALIZEDTIME) {
-        memset(tm, 0, sizeof(*tm));
-        return asn1_generalizedtime_to_tm(tm, s);
-    }
-
-    return 0;
+    memset(tm, 0, sizeof(*tm));
```

Comment:
One can ask same question whether or not it's appropriate to modify `*tm` if asn1_time_to_tm() returns error. Why not simply let asn1_time_to_tm() decide? There is certain ambiguity in documentation, it says "Only the B<tm_sec>, B<tm_min>, B<tm_hour>, B<tm_mday>, B<tm_mon> and B<tm_year> fields are updated." It sounds like that if you pass a tm if won't touch fields other than listed. This doesn't seem to be true. I mean it wasn't true to start with... Hmmm...